src/clr-core/modal/modal.element.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  applyMixins,
+  baseStyles,
+  CommonStringsService,
+  CssHelpers,
+  ESC,
+  event,
+  EventEmitter,
+  property,
+  registerElementSafely,
+  UniqueId,
+} from '@clr/core/internal';
+import { html } from 'lit-element';
+import { CdsBaseFocusTrap } from '../internal/base/focus-trap.base';
+import { styles } from './modal.element.css';
+
+class ModalMixinClass extends CdsBaseFocusTrap {}
+
+applyMixins(ModalMixinClass, [UniqueId, CssHelpers]);
+
+/**
+ * Web component modal.
+ *
+ * ```typescript
+ * import '@clr/core/modal';
+ * ```
+ *
+ * ```html
+ * <cds-modal>
+ *   <cds-modal-header>
+ *      <h3 cds-text="title">My Modal</h3>
+ *   </cds-modal-header>
+ *   <cds-modal-content>
+ *     <p>Lorem Ipsum</p>
+ *   </cds-modal-content>
+ *   <cds-modal-actions>
+ *       <cds-button>Ok</cds-button>
+ *   </cds-modal-actions>
+ * </cds-modal>
+ * ```
+ * @beta
+ * @element cds-modal
+ */
+export class CdsModal extends ModalMixinClass {
+  static get styles() {
+    return [baseStyles, styles];
+  }
+  @event() private closeChange: EventEmitter<boolean>;
+
+  /** If false, the modal will not render the close button.  */
+  @property({ type: Boolean })
+  closable = true;
+
+  /** Sets the overall height and width of the modal and icon based on value */
+  @property({ type: String })
+  size: 'default' | 'sm' | 'lg' | 'xl';
+
+  private idForAriaLabel = `${this._idPrefix}${this._uniqueId}`;
+
+  render() {
+    return html`
+      <div class="private-host" cds-layout="horizontal p:sm p@md:lg align:center">
+        <div class="modal-dialog" 
+              role="dialog" 
+              aria-modal="true" 
+              aria-labelledby="${this.idForAriaLabel}">
+          <div cds-layout="display:screen-reader-only">${CommonStringsService.keys.modalContentStart}</div>
```

Comment:
All of these commonstrings will need to be replaced by the end user to something meaningful and unique to the modal. I don't think we want to handle i18n the same way in core as we do in angular. the strategy is to provide some default strings using commonstrings and warn end users that if they don't switch them out then they risk not being accessible.

because all of these sr-only and aria values are always and only strings, we could make them properties/attributes.

Reply:
Yes, I believe the consensus was to address this separately after this PR? 😁

Reply:
maybe... it _is_ Wednesday. so things might be getting foggy for me...

Reply:
Yeah we decided to keep that a separate PR to work through and apply to all components. We may need a hybrid of the two but its still undecided how or what the API may look like. I was still planning on opening an issue with a couple of ideas we could discuss. 

```typescript
@property({ type: String, { required: 'warning', requiredMessage: 'must be set for a11y' } })
closeLabel = CommonStringsService.keys.close;
```

```html
<cds-modal close-label="Close Sign Up Modal">
  <h1>Sign Up</h1>
</cds-modal>
```